test-d/non-empty-string.ts

```diff
@@ -0,0 +1,11 @@
+import {expectNever, expectType} from 'tsd';
+import type {NonEmptyString} from '../index';
+
+declare const empty: NonEmptyString<''>;
+expectNever(empty);
```

Comment:
Better to use `expectType` instead of `expectNever`.

```suggestion
declare const empty: NonEmptyString<''>;
expectType<never>(empty);
```